internal/identity/identity.go

```diff
@@ -13,7 +13,11 @@ type Principal = identity.XRHID
 
 // Identity returns identity header struct or nil when not set.
 func Identity(ctx context.Context) Principal {
-	return identity.Get(ctx)
+	val := ctx.Value(identity.Key)
+	if val == nil {
+		return Principal{}
+	}
+	return val.(Principal)
```

Comment:
The `Get` from middleware project does not handle `nil` correctly, so this is a copy.